tests/neg-custom-args/isInstanceOf/i5826.scala

```diff
@@ -1,6 +1,6 @@
 class Foo {
   def test[A]: List[Int] | A => Int = {
-    case ls: List[Int] => ls.head       // error
+    case ls: List[Int] => ls.head       // ok, List decomposes to Some and None
```

Comment:
what if `A` is `List[String]`? shouldn't this be unchecked?

Reply:
Yes, that would be unchecked --- we already have tests for that in this file.

Reply:
right, but I see that `-Xfatal-warnings` is turned on for this test, but this line does not produce the unchecked warning

Reply:
On my local machine, I get the following, which is as expected:

```Scala
-- Unchecked Warning: tests/neg-custom-args/isInstanceOf/i5826.scala:8:9 -------
8 |    case ls: List[Int] => ls.head       // error
  |         ^
  |the type test for List[Int] cannot be checked at runtime because it cannot be checked against the type List[String]
-- Unchecked Warning: tests/neg-custom-args/isInstanceOf/i5826.scala:17:9 ------
17 |    case ls: A[X] => 4                 // error
   |         ^
   |the type test for Foo.this.A[X] cannot be checked at runtime because it cannot be checked against the type Foo.this.B[X]
2 warnings found
```

Reply:
`test[List[String]](List("a"))` would crash

Reply:
> test[List[String]](List("a")) would crash

It will not, because the other is a function type `A => Int`, not `A`.

Reply:
Sorry, I see I misunderstood twice (and they don't cancel).  `List[Int] | A => Int` is `(List[Int] | A) => Int`, thus a warning is expected here.

We need to fix this test.

Reply:
But it crashed when I tried it?

Reply:
ok, I saw your comment now :)